### PR TITLE
External Gluster File ( gluster.org/glusterfile ) provisioner.

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: gluster/file/deploy/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  replicas: 1
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: glusterfile-provisioner
+    spec:
+      containers:
+        -
```
Comment (rootfs) [inline gluster/file/deploy/deployment.yaml]: remove blank lines (here and other yamls)

File: gluster/file/deploy/clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1alpha1
```
Comment (rootfs) [inline gluster/file/deploy/clusterrole.yaml]: v1

File: gluster/file/deploy/clusterrolebinding.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1alpha1
```
Comment (rootfs) [inline gluster/file/deploy/clusterrolebinding.yaml]: v1

File: gluster/file/deploy/clusterrole.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["list", "watch", "create", "update", "patch"]
+  - apiGroups: [""]
+    resources: ["services"]
+    verbs: ["get"]
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "create", "delete"]
```
Comment (rootfs) [inline gluster/file/deploy/clusterrole.yaml]: why need delete?

File: gluster/file/deploy/clusterrole.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    verbs: ["get", "list", "watch", "update"]
+  - apiGroups: ["storage.k8s.io"]
+    resources: ["storageclasses"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["list", "watch", "create", "update", "patch"]
+  - apiGroups: [""]
+    resources: ["services"]
+    verbs: ["get"]
```
Comment (rootfs) [inline gluster/file/deploy/clusterrole.yaml]: no create and delete?

**Comment by rootfs** [comment]
please refine rbac rules and expose less cluster roles.

**Comment by rootfs** [comment]
update header
```
Boilerplate header is wrong for: /home/travis/gopath/src/github.com/kubernetes-incubator/external-storage/gluster/file/cmd/glusterfile-provisioner/glusterfile-provisioner.go
```

File: gluster/file/deploy/clusterrolebinding.yaml (hunk 3 above)
Comment (humblec) [reply gluster/file/deploy/clusterrolebinding.yaml]: Done

File: gluster/file/deploy/clusterrole.yaml (hunk 4 above)
Comment (humblec) [reply gluster/file/deploy/clusterrole.yaml]: Done.

File: gluster/file/deploy/clusterrole.yaml (hunk 5 above)
Comment (humblec) [reply gluster/file/deploy/clusterrole.yaml]: Added

**Comment by humblec** [comment]
@rootfs Thanks, addressed the comments.

File: gluster/file/deploy/clusterrole.yaml (hunk 2 above)
Comment (humblec) [reply gluster/file/deploy/clusterrole.yaml]: Done

**Comment by rootfs** [comment]
thanks @humblec , let's have it now and improve. 

/lgtm

@childsb can you merge? thanks

File: gluster/block/Dockerfile [hunk 6]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2017 The Kubernetes Authors.
+# Copyright 2018 The Kubernetes Authors.
```
[the remark this answers is not in the document]
Comment (humblec) [reply gluster/block/Dockerfile]: Removed it from this commit.

File: gluster/file/CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,4 @@
+# v0.0
+- Initial drop of Gluster File Provisioner.
+- Humble Chirammal <hchirmam@redhat.com>
```
[the remark this answers is not in the document]
Comment (humblec) [reply gluster/file/CHANGELOG.md]: Corrected the email address, changelog helps to version the binary in different stages.

File: gluster/file/README.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
+# glusterfile Volume Provisioner for Kubernetes 1.5+
+
+[![Docker Repository on Quay](https://quay.io/repository/external_storage/glusterfile-provisioner/status "Docker Repository on Quay")](https://quay.io/repository/external_storage/glusterfile-provisioner)
+```
+quay.io/external_storage/glusterfile-provisioner:latest
+```
+## What is Gluster File Provisioner ?
+
+Gluster File Provisioner is an external provisioner which dynamically provision gluster file volumes  on demand. The persistent Volume Claim which has been requested with this external provisioner's identity ( for eg# `gluster.org/glusterfile`)  will be served by this provisioner.
```
[the remark this answers is not in the document]
Comment (humblec) [reply gluster/file/README.md]: Done.

[the remark this answers is not in the document]
Comment (humblec) [reply gluster/file/README.md]: Done.

File: gluster/file/README.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+[![Docker Repository on Quay](https://quay.io/repository/external_storage/glusterfile-provisioner/status "Docker Repository on Quay")](https://quay.io/repository/external_storage/glusterfile-provisioner)
+```
+quay.io/external_storage/glusterfile-provisioner:latest
+```
+## What is Gluster File Provisioner ?
+
+Gluster File Provisioner is an external provisioner which dynamically provision gluster file volumes  on demand. The persistent Volume Claim which has been requested with this external provisioner's identity ( for eg# `gluster.org/glusterfile`)  will be served by this provisioner. 
+
+[glusterfile](https://github.com/gluster/glusterfile)
```
[the remark this answers is not in the document]
Comment (humblec) [reply gluster/file/README.md]: Corrected.

**Comment by humblec** [comment]
@obnoxxx addressed your comments!

**Comment by humblec** [comment]
@obnoxxx please let me know if you have any further comments.

**Pull request merged**